persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -207,14 +227,9 @@ private void authorizeByToken(ByteBuffer token, CQLStatement statement) {
           castStatement.keyspace(),
           castStatement.columnFamily());
     } else if (statement instanceof ModificationStatement) {
-      ModificationStatement castStatement = (ModificationStatement) statement;
-      Scope scope;
-      if (statement instanceof DeleteStatement) {
-        scope = Scope.DELETE;
-      } else {
-        scope = Scope.MODIFY;
-      }
-
+      authorizeModificationStatement(statement, authToken, authorization);
+    } else if (statement instanceof TruncateStatement) {
+      TruncateStatement castStatement = (TruncateStatement) statement;
```

Comment:
Why do we need this special case? Previously, it was working for all scopes, and now we have a dedicated branch for `TRUNCATE`. Shouldn't both use the generic mechanism?

Reply:
There was always a dedicated branch for truncate, this diff just looks weird after I extracted a method for ModificationStatement